c/authService.c

```diff
@@ -173,6 +183,116 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
   return 0;
 }
 
+void respondWithJsonStatus(HttpResponse *response, const char *status, int statusCode, const char *statusMessage) {
+    jsonPrinter *out = respondWithJsonPrinter(response);
+    setResponseStatus(response,statusCode,(char *)statusMessage);
+    setDefaultJSONRESTHeaders(response);
+    writeHeader(response);
+
+    jsonStart(out);
+    jsonAddString(out, "status", (char *)status);
+    jsonEnd(out);
+
+    finishResponse(response);
+}
+
+static int resetPassword(HttpService *service, HttpResponse *response) {
+  int returnCode = 0, reasonCode = 0;
+  HttpRequest *request = response->request;
+  
+  if (!strcmp(request->method, methodPOST)) {
+    char *inPtr = request->contentBody;
+    char *nativeBody = copyStringToNative(request->slh, inPtr, strlen(inPtr));
+    int inLen = nativeBody == NULL ? 0 : strlen(nativeBody);
+    char errBuf[JSON_ERROR_BUFFER_SIZE];
+    char responseString[RESPONSE_MESSAGE_LENGTH];
+
+    if (nativeBody == NULL) {
+      respondWithJsonStatus(response, "No body found", HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    }
+
+    Json *body = jsonParseUnterminatedString(request->slh, nativeBody, inLen, errBuf, JSON_ERROR_BUFFER_SIZE);
+    JsonObject *inputMessage = jsonAsObject(body);
+    Json *username = jsonObjectGetPropertyValue(inputMessage,"username");
+    Json *password = jsonObjectGetPropertyValue(inputMessage,"password");
+    Json *newPassword = jsonObjectGetPropertyValue(inputMessage,"newPassword");
+    int usernameLength = 0;
+    int passwordLength = 0;
+    int newPasswordLength = 0;
+    if (username != NULL) {
+      if (jsonAsString(username) != NULL) {
+        usernameLength = strlen(jsonAsString(username));
+      }
+    }
+    if (password != NULL) {
+      if (jsonAsString(password) != NULL) {
+        passwordLength = strlen(jsonAsString(password));
+      }
+    }
+    if (newPassword != NULL) {
+      if (jsonAsString(newPassword) != NULL) {
+        newPasswordLength = strlen(jsonAsString(newPassword));
+      }
+    }
+    if (usernameLength == 0) {
+      respondWithJsonStatus(response, "No username provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else if (passwordLength == 0) {
+      respondWithJsonStatus(response, "No password provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else if (newPasswordLength == 0) {
+      respondWithJsonStatus(response, "No new password provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else {
+      resetZosUserPassword(jsonAsString(username),  jsonAsString(password), jsonAsString(newPassword), &returnCode, &reasonCode);
+
+      if (returnCode == PASSWORD_RESET_RC_OK) {
+        respondWithJsonStatus(response, "Password Successfully Reset", HTTP_STATUS_OK, "OK");
```

Comment:
If you changed this after my comment about `if-else` then you must have misinterpreted it. I didn't mean removing the switch case. I meant making the passwordLength, newPasswordLength, etc checks just normal `if` check without else since they return immediately. The you could also remove the `else` on line 250 making it more readable.

Anyway, it's fine.